easy-connect.cpp

```diff
@@ -237,6 +245,31 @@ NetworkInterface* easy_connect(bool log_messages) {
 #  endif
     connect_success = cellular.connect();
     network_interface = &cellular;
+
+#elif MBED_CONF_APP_NETWORK_INTERFACE == CELLULAR_WNC14A2A
+    if (log_messages) {
+        printf("[EasyConnect] Using WNC14A2A\n");
+    }
+#   if MBED_CONF_APP_WNC_DEBUG == true
+        printf("[EasyConnect] With WNC14A2A debug output set to 0x%02X\n",MBED_CONF_APP_WNC_DEBUG_SETTING);
+        wnc = new WNC14A2AInterface(&dbgout);
+        if (wnc == NULL) {
```

Comment:
All the other devices are using statically allocated objects, is there any reason why this is not following the same style?

Reply:
The primary reason this object is using dynamic allocation vs. static is due to mbed o/s legacy issues.  Originally, the driver was written for mbed os 2.x.  There was no intrinsic UART support support provided in the os.  As a result, you had to use objects such as BufferedSerial or  MODSERIAL to handle the UART interaction.  Because the data exchanges with the WNC part required large buffers (4K).  The object wasn’t built on the stack (statically allocated) because the stack requirements for this were excessive and caused problems in mbed.  There was plenty of space on the HEAP for dynamic allocation and this eliminated all the problems. 

As the mbed os has evolved, it now is supporting UARTs via UARTSerial and modifications of the buffer size is possible via the mbed_app.json file.  This is a good addition but wasn’t added until only after mbed v5.x, well after the wnc object was designed. 

My question is just because this object is allocated differently than the others, is there a compelling reason to redesign it?

Reply:
The static objects are not consuming stack space, they are living in a separate region, statically allocated at linking time. Unless the WNC14A2AInterface's constructor is relying on some specific static initialization order, I don't believe there is any need for redesign, just a minor change in the integration here.

One could have this kind of block in the around line 100 
```
#elif MBED_CONF_APP_NETWORK_INTERFACE == CELLULAR_WNC14A2A
#include "WNC14A2AInterface.h"
#if MBED_CONF_APP_WNC_DEBUG == true
#include "WNCDebug.h"
WNCDebug dbgout(stderr);
WNC14A2AInterface wnc(&dbgout);
#else
WNC14A2AInterface wnc();
#endif
#else
```

and in the easy_connect()
```
#elif MBED_CONF_APP_NETWORK_INTERFACE == CELLULAR_WNC14A2A
    if (log_messages) {
        printf("[EasyConnect] Using WNC14A2A\n");
    }
#if MBED_CONF_APP_WNC_DEBUG == true
    printf("[EasyConnect] With WNC14A2A debug output set to 0x%02X\n", MBED_CONF_APP_WNC_DEBUG_SETTING);
    wnc.doDebug(MBED_CONF_APP_WNC_DEBUG_SETTING);
#endif
    network_interface = &wnc;
    connect_success = wnc.connect();
#else
```